pkg/image/utilities/utilities.go

```diff
@@ -622,52 +653,67 @@ if ($help)
     exit 0
 }
 
-if (-not $registry)
+if (-not $target_registry)
 {
-	echo "Registry address is required"
+    echo "Registry address is required"
     usage
     exit 1
 }
 
 if (-not (Test-Path $images))
 {
-	echo "Could not find '$images'"
+    echo "Could not find '$images'"
     usage
-	exit 1
+    exit 1
 }
 
 docker load --input $images
 if (-not $?)
 {
-	echo "Could not load '$images'"
+    echo "Could not load '$images'"
     exit 1
 }
 
 if (-not (Test-Path $image_list))
 {
-	exit 0
+    exit 0
+}
+
+$target_registry = $target_registry.TrimEnd("/") + "/"
+
+if ([string]::IsNullOrEmpty($source_registry))
+{
+    $source_registry = ""
+}
+else
+{
+    $source_registry = $source_registry.TrimEnd("/") + "/"
 }
 
 Get-Content -Force -Path $image_list | ForEach-Object {
     if ($_) {
         $fullname_image = ('{0}-windows-{1}' -f $_, $os_release_id)
-		echo "Tagging $registry/$fullname_image"
-	
-		switch -regex ($fullname_image)
-		{
-			'.+/.+' {
-				docker tag $fullname_image $registry/$fullname_image
+        $source_image = -join ($source_registry, $fullname_image)
+
+        switch -regex ($fullname_image)
+        {
+            '.+/.+' {
+                $target_image = -join ($target_registry, $fullname_image)
+                echo "Tagging $target_image"
+                docker tag $source_image $target_image
                 if ($?) {
-                    docker push $registry/$fullname_image
+                    docker push $target_image
                 }
-			}
-			default {
-				docker tag $fullname_image $registry/rancher/$fullname_image
-				if ($?) {
-                    docker push $registry/rancher/$fullname_image
+            }
+            default {
+                $target_image = -join ($target_registry, "rancher/", $fullname_image)
+                echo "Tagging $target_image"
+                docker tag $source_image $target_image
```

Comment:
does `$source_image` here also need to be modified to `-join ($source_registry, "rancher/", $fullname_image)`?

Reply:
In this case, the piece that the switch acts upon is already a part of the `$source_image` content:

https://github.com/rancher/rancher/blob/7eff43334bef7fa2423d611a14ed4d33dffeddc5/pkg/image/utilities/utilities.go#L696-L699

Reply:
It cannot be modified as the original image reference already has the `rancher/` part (or not). Hence the `switch` statement to determine whether it should be tagged with `rancher/` or not.

Reply:
yes this makes sense, this seems to be how the script has been built for linux as well.

Reply:
Updating this discussion: the use case here is most likely the situation where an image is coming from a Docker Hub library repository, but we need to push it to a private repository (which will most likely require a project to do so).